src/Slideshow.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { filmData } from "./film-data";
+
+const Slideshow = () => {
+    const [filmData] = useState(filmData);
+    const [index, setIndex] = useState(0);
+
+    useEffect(() => {
+        const lastIndex = filmData.length - 1;
+        if(index < 0) {
+            setIndex(lastIndex);
+        }
+        if(index > lastIndex) {
+            setIndex(0);
+        }
+    }, [index, filmData])
+
+    useEffect(() => {
+        const nextIndex = filmData.length + 1;
+        if(index < 0) {
+            setIndex(nextIndex);
+        }
+        if(index > nextIndex) {
+            setIndex(0);
+        }
+    }, [index, filmData])
+    
```

Comment:
What if you took the logic for changing the index and put it instead closer to where the action is, where you have your buttons? Or you could write a click handler function that is called when buttons are clicked. For example, if the next button is clicked you run a function that updates the state of `index` by adding 1 to its value.